apiserver/test/e2e/apiserversdk/exec.go

```diff
@@ -34,3 +52,169 @@ func (rt *ProxyRoundTripper) RoundTrip(req *http.Request) (*http.Response, error
 
 	return rt.Transport.RoundTrip(newReq)
 }
+
+// RemoteExecuteClient allows executing HTTP requests against a service running inside a Kubernetes pod
+// by using `kubectl exec`-style command execution, without requiring a NodePort for external access.
+type RemoteExecuteClient struct {
```

Comment:
Could we use the existing ProxyRoundTripper instead of introducing a RemoteExecuteClient?